pyboy/plugins/debug.py

```diff
@@ -522,6 +539,119 @@ def update_title(self):
+                sdl2.SDL_RenderCopy(self._sdlrenderer, self.texture, None, self.destination)
+                self.row_index += 18
+            
+    def body(self, row_index):
+            self.surface = TTF_RenderText_Solid(self.font, self.row.encode(), self.text_color)
+            self.texture = sdl2.SDL_CreateTextureFromSurface(self._sdlrenderer, self.surface)
+            self.get_destination(self.row_index)
+            sdl2.SDL_RenderCopy(self._sdlrenderer, self.texture, None, self.destination)
+            self.address_index += 0x8
+            self.row_index += 18
+
+    def footer(self, row_index):
+            self.row = "-" * 37
+            self.surface = TTF_RenderText_Solid(self.font, self.row.encode(), self.text_color)
+            self.texture = sdl2.SDL_CreateTextureFromSurface(self._sdlrenderer, self.surface)
+            self.get_destination(self.row_index)
+            sdl2.SDL_RenderCopy(self._sdlrenderer, self.texture, None, self.destination)
+
+    def get_destination(self, row_index):
+            self.destination = sdl2.SDL_Rect(int(0), int(row_index))
+            w = ctypes.pointer(ctypes.c_int(0))
+            h = ctypes.pointer(ctypes.c_int(0))
+            sdl2.SDL_QueryTexture(self.texture, None, None, w, h)
+            self.destination.w = w.contents.value
+            self.destination.h = h.contents.value
+
+    def handle_events(self, events):
+        events = BaseDebugWindow.handle_events(self, events)
+        for event in events:
+            # j - Next 256 bytes
+            if event == WindowEvent.PRESS_BUTTON_J:
+                if self.end_address + 0x100 <= 0x10000:
+                    self.start_address += 0x100
+                    self.end_address += 0x100
+            # k - Last 256 bytes
+            elif event == WindowEvent.PRESS_BUTTON_K:
+                if self.start_address - 0x100 >= 0x0000:
+                    self.start_address -= 0x100
+                    self.end_address -= 0x100
+            # left shift - Forward 4096 bytes
+            elif event == WindowEvent.PRESS_BUTTON_LSHIFT:
+                if self.end_address + 0x1000 <= 0x10000:
+                    self.start_address += 0x1000
+                    self.end_address += 0x1000
+                else:
+                    self.start_address = 0xff00
+                    self.end_address = 0x10000
+            # right shift - Back 4096 bytes
+            elif event == WindowEvent.PRESS_BUTTON_RSHIFT:
+                if self.start_address - 0x1000 >= 0x0000:
+                    self.start_address -= 0x1000
+                    self.end_address -= 0x1000
+                else:
+                    self.start_address = 0x0000
+                    self.end_address = 0x100
+
+        return events
+
+    def draw_overlay(self):
+        pass
+
+    def copy_tile(self, tile_cache0, t, xx, yy, to_buffer):
+        pass
+
+    def mark_tile(self, x, y, color, height, width, grid):
+        pass
```

Comment:
Did you need to define these?

Reply:
I believe i did that to just override the default behavior with a pass so it wouldn't do anything wonky.